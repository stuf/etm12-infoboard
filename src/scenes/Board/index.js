import * as React from "karet";

export default function Board(props) {
  return <section>Main board</section>;
}
