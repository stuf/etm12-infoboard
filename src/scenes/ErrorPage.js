import * as React from "karet";

export function NotFound(props) {
  return <section>Not found :(((</section>;
}
