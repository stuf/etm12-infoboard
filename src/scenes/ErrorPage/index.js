import * as React from 'karet';
import * as U from 'karet.util';

export function NotFound(props) {
  return <section>Not found :(((</section>;
}

export function InvalidState(props) {
  const { state } = props;

  return (
    <section>
      <h1>Oh no!</h1>

      <h2>Invalid state data</h2>

      <p>
        It seems as the state the application is in is not valid. It's
        currently:
      </p>

      <pre>
        <code>{U.stringify(state, null, 2)}</code>
      </pre>
    </section>
  );
}
