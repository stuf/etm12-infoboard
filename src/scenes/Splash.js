import * as React from "karet";
import * as U from "karet.util";

export default function Index(props) {
  return (
    <section>
      Index page
      <pre>
        <code>{U.stringify(props, null, 2)}</code>
      </pre>
    </section>
  );
}
