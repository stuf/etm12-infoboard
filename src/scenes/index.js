import * as ErrorPage from "./ErrorPage";

export { ErrorPage };

export { default as Splash } from "./Splash";
export { default as Board } from "./Board";
