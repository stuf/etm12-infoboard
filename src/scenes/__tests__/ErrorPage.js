import * as React from 'karet';
import * as U from 'karet.util';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import * as ErrorPage from 'scenes/ErrorPage';

describe('Error pages', () => {
  describe('NotFound', () => {
    it('matches snapshot', () => {
      const wrapper = mount(<ErrorPage.NotFound />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('InvalidState', () => {
    it('matches snapshot', () => {
      const wrapper = mount(<ErrorPage.InvalidState />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
