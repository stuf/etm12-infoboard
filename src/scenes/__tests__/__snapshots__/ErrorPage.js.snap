// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error pages InvalidState matches snapshot 1`] = `
<InvalidState>
  <section>
    <h1>
      Oh no!
    </h1>
    <h2>
      Invalid state data
    </h2>
    <p>
      It seems as the state the application is in is not valid. It's currently:
    </p>
    <pre>
      <code />
    </pre>
  </section>
</InvalidState>
`;

exports[`Error pages NotFound matches snapshot 1`] = `
<NotFound>
  <section>
    Not found :(((
  </section>
</NotFound>
`;
